pkg/resizer/csi_resizer.go

```diff
@@ -148,7 +176,12 @@ func (r *csiResizer) Resize(pv *v1.PersistentVolume, requestSize resource.Quanti
 	if err != nil {
 		return oldSize, nodeResizeRequired, err
 	}
-	return *resource.NewQuantity(newSizeBytes, resource.BinarySI), nodeResizeRequired, err
+
+	if pv.Spec.VolumeMode == nil || *pv.Spec.VolumeMode == v1.PersistentVolumeFilesystem {
+		return *resource.NewQuantity(newSizeBytes, resource.BinarySI), nodeResizeRequired, err
+	}
+
+	return *resource.NewQuantity(newSizeBytes, resource.BinarySI), false, err
```

Comment:
Are there any block plugins that only implement node resize?

Reply:
Not that Im aware of, could it be a case for local volume plugin? And how is this going to affect this highlighted line?

Reply:
There is nothing in the csi spec that limits node expansion to only filesystem volumes.  For a block plugin that only supports node resize, then is setting nodeResizeRequired=false going to cause nodeResize to never be called?

Should we instead always call node expansion (based off of nodeResizeRequired)?  If the plugin gets a node expand call for a block device, can they just return success immediately?

Reply:
I agree that the responsibility of determining whether node expansion is needed or not should be delegated to the plugin, but in control-plane the plugin may not have the necessary information to know that volume is being used in block device mode. 

For now - I think it makes sense to return whatever `nodeResizeRequired` value was returned by the plugin. In future I am thinking we may want to update CSI spec to pass volume capability with `ControllerExpandVolume` request.

Reply:
I think the question here is more related to node expand.  It looks like we don't pass volume capability to node expansion, and maybe we need to add that?

I guess what a plugin could do now is check if the volume has a filesystem on it. If not, then assume it's raw block, otherwise assume it's filesystem.

Reply:
Shouldn't a plugin be able to determine that given `target_path` in `NodeExpandVolume` RPC call is a device or a mounted filesystem?

Reply:
agree with @gnufied here, `ControllerExpand` should have volume capability in the future so the driver can determine `nodeResizeRequired` better.

`NodeExpandVolume` may be determinable by the `target_path` but it might be nice to know the volume capability too.

Either way, for this PR I think just returning the plugins `nodeResizeRequired` makes sense instead of overwriting with `false`.

Reply:
Unsure if we should be dynamically determining whether or not to call node expand. It is simpler to base it strictly on the plugin's capabilities. Especially across controller and node services, this makes testing/supporting version skew more complicated.